File: Examples/Examples/ListView/LoremIpsumGenerator.cs
using System;
using System.Collections.Generic;
using System.Text;

using Foundation;

namespace Examples
{
	[Register("LoremIpsumGenerator")]
	public class LoremIpsumGenerator
	{
		string[] words = new string[] {"lorem", "ipsum", "dolor", "sit", "amet", "consectetuer", "adipiscing", "elit", "integer", "in", "mi", "a", "mauris"};
		Dictionary<NSIndexPath, string> rows = new Dictionary<NSIndexPath, string> ();

		public string GenerateString(int wordCount)
		{
			StringBuilder str = new StringBuilder ();
			Random r = new Random ();
			for (int i = 0; i < wordCount; i++) {
				str.Append (words [r.Next(0, words.Length)]);
				str.Append (" ");
			}

			return str.ToString ();
		}

		public string RandomString(int wordCount, NSIndexPath indexPath)
		{
			string text = null;
			if (rows.ContainsKey (indexPath)) {
				text = rows [indexPath];
			} else {
				text = this.GenerateString (wordCount);
				rows [indexPath] = text;
			}

			return text;
		}
	}
}

File: TelerikUIExamplesInSwift/TelerikUIExamplesInSwift/LoremIpsumGenerator.swift
//
//  LoremIpsumGenerator.swift
//  TelerikUIExamplesInSwift
//
//  Copyright (c) 2015 Telerik. All rights reserved.
//

import UIKit

class LoremIpsumGenerator: NSObject {
    
    let words = ["lorem", "ipsum", "dolor", "sit", "amet", "consectetuer", "adipiscing", "elit", "integer", "in", "mi", "a", "mauris"]
    let rows = NSMutableDictionary()
    
    func generateString(wordCount: NSInteger) -> NSString {
        let randomString = NSMutableString()
        for _ in 0 ..< wordCount {
            let index : Int = Int(arc4random_uniform(UInt32(words.count)))
            randomString.appendString(words[index])
            randomString.appendString(" ")
        }
        return randomString
    }
    
    func randomString(wordCount: NSInteger, indexPath: NSIndexPath) -> NSString {
        var text : NSString? = rows.objectForKey(indexPath) as? NSString
        if(text == nil){
          text = generateString(wordCount)
          rows.setObject(text!, forKey: indexPath)
        }
        return text!
    }
}
